### ISSUE TITLE
utf8 decode of malformed string is different from node

### ISSUE THREAD

**Issue by jakebailey** [oven-sh/bun#8372]
### What version of Bun is running?

1.0.25+a8ff7be64

### What platform is your computer?

Linux 5.15.90.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

The file that I caught this on was an MPEG Transport Stream file I added to TS as a test.

```js
import { createHash } from "crypto";
import { StringDecoder } from "string_decoder";

function hash(text) {
  const hash = createHash("sha256");
  hash.update(text);
  return hash.digest("hex");
}

const buffer = Buffer.from("R0ABkgEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQE=", "base64");
const array = new Uint8Array(buffer);

console.log(`hash of data: ${hash(array)}`);

const decoder = new StringDecoder("utf-8");
const text = decoder.write(array);

console.log(`hash of text: ${hash(text)}`);
console.log(`text: ${text}`);
```

### What is the expected behavior?

```console
$ node index.mjs
hash of data: 8c3e76219ed3d50ebe8f47bcbbb58201f765d4445c618b64ab41e3a85aa4044a
hash of text: 2637e4a587f78c8be9f18fec1ad0fcbe5ee9ca90e01c7c9ce61a7fe6ab7d6610
text: G@�
```

### What do you see instead?

```console
$ bun index.mjs
hash of data: 8c3e76219ed3d50ebe8f47bcbbb58201f765d4445c618b64ab41e3a85aa4044a
hash of text: 5288af01e8b8c3f6cca36506260d8e9a91b20d8c85f1858fce774a4f3ae5ab50
text: G@R�
```

Note the `R` (though looking at it as text is not the best way to observe the change).

### Additional information

This breaks TS tests (once working after #7664) as bun appears to decode malformed utf-8 differently.

**Comment by jakebailey** [comment]
I'll note that I used a Buffer and StringDecoder as they were easy for an example, but this happens for fetch's `Response#text()` and `fs.readFile(..., "utf8")` too (the latter is what TS uses).

**Comment by Jarred-Sumner** [comment]
Fixed in Bun v1.0.26

```ts
❯ bun-1.0.26 /tmp/node-a.mjs
hash of data: 8c3e76219ed3d50ebe8f47bcbbb58201f765d4445c618b64ab41e3a85aa4044a
hash of text: 2637e4a587f78c8be9f18fec1ad0fcbe5ee9ca90e01c7c9ce61a7fe6ab7d6610
text: G@�

bun on  main 
❯ bun-1.0.27 /tmp/node-a.mjs
hash of data: 8c3e76219ed3d50ebe8f47bcbbb58201f765d4445c618b64ab41e3a85aa4044a
hash of text: 2637e4a587f78c8be9f18fec1ad0fcbe5ee9ca90e01c7c9ce61a7fe6ab7d6610
text: G@�

bun on  main 
❯ bun-1.0.28 /tmp/node-a.mjs
hash of data: 8c3e76219ed3d50ebe8f47bcbbb58201f765d4445c618b64ab41e3a85aa4044a
hash of text: 2637e4a587f78c8be9f18fec1ad0fcbe5ee9ca90e01c7c9ce61a7fe6ab7d6610
text: G@�

bun on  main 
❯ bun /tmp/node-a.mjs
hash of data: 8c3e76219ed3d50ebe8f47bcbbb58201f765d4445c618b64ab41e3a85aa4044a
hash of text: 2637e4a587f78c8be9f18fec1ad0fcbe5ee9ca90e01c7c9ce61a7fe6ab7d6610
text: G@�

bun on  main 
❯ node /tmp/node-a.mjs
hash of data: 8c3e76219ed3d50ebe8f47bcbbb58201f765d4445c618b64ab41e3a85aa4044a
hash of text: 2637e4a587f78c8be9f18fec1ad0fcbe5ee9ca90e01c7c9ce61a7fe6ab7d6610
text: G@�
```
